src/main/asciidoc/_chapters/hbase_mob.adoc

```diff
@@ -181,84 +320,51 @@ suit your environment, and restart or rolling restart the RegionServer.
-size. When memstores from a column family configured to use MOB are eventually flushed two hfiles
-are written simultaneously. Cells with a value smaller than the threshold size are written to a
-normal region hfile. Cells with a value larger than the threshold are written into a special MOB
-hfile and also have a MOB reference cell written into the normal region HFile.
-
-MOB reference cells have the same key as the cell they are based on. The value of the reference cell
-is made up of two pieces of metadata: the size of the actual value and the MOB hfile that contains
-the original cell. In addition to any tags originally written to HBase, the reference cell prepends
-two additional tags. The first is a marker tag that says the cell is a MOB reference. This can be
-used later to scan specifically just for reference cells. The second stores the namespace and table
-at the time the MOB hfile is written out. This tag is used to optimize how the MOB system finds
-the underlying value in MOB hfiles after a series of HBase snapshot operations (ref HBASE-12332).
-Note that tags are only available within HBase servers and by default are not sent over RPCs.
+This same request can be made via the `Admin.majorCompact` Java API.
 
-All MOB hfiles for a given table are managed within a logical region that does not directly serve
-requests. When these MOB hfiles are created from a flush or MOB compaction they are placed in a
-dedicated mob data area under the hbase root directory specific to the namespace, table, mob
-logical region, and column family. In general that means a path structured like:
+=== MOB Troubleshooting
 
-----
-%HBase Root Dir%/mobdir/data/%namespace%/%table%/%logical region%/%column family%/
-----
+==== Adjusting the MOB cleaner's tolerance for new hfiles
 
-With default configs, an example table named 'some_table' in the
-default namespace with a MOB enabled column family named 'foo' this HDFS directory would be
+The MOB cleaner chore ignores all MOB hfiles that were created more recently than an hour prior to
+the start of the shore to ensure we don't miss the reference metadata from teh corresponding regular
+hfile. Without this safety check it would be possible for the cleaner chore to see a MOB hfile for
+an in progress flush or compaction and prematurely archive the MOB data. This default buffer should
+be sufficient for normal use.
 
-----
-/hbase/mobdir/data/default/some_table/372c1b27e3dc0b56c3a031926e5efbe9/foo/
-----
-
-These MOB hfiles are maintained by special chores in the HBase Master rather than by any individual
-Region Server. Specifically those chores take care of enforcing TTLs and compacting them. Note that
-this compaction is primarily a matter of controlling the total number of files in HDFS because our
-operational assumptions for MOB data is that it will seldom update or delete.
+You will need to adjust the tolerance if it takes longer than an hour for the two HDFS move
```

Comment:
The algorithm is currently slightly different. When compaction chore starts I record start time = startTime, then calculate 

maxCreationTimeToArchive = startTime - minAgeToArchive

so, obviously, 1h lagging is more than enough to avoid any possible race conditions and now we do not depend on compaction chore running time.

Reply:
the paragraph doesn't talk about the time to do things on the master in the cleaning chore. It's talking about the amount of time it takes the region server that's doing a flush or compaction.

for example, consider if something is *very* wrong with HDFS such that the RS doing a flush commits the mob file and then pauses for > 1 hour before committing the reference hfile. if the master did it's "min age" calculation an hour after the mob file commit and it finished iterating over the reference files prior to the commit of the reference hfile, then the cleaner chore would believe it is fine to delete the mob hfile.

similarly, if the compaction process commits a mob hfile (due to the size limitation) and then spends another 3 hours writing other mob hfiles before finally closing out and committing the reference hfile, then the master cleaner chore could have the timing work out so that some of the early mob hfiles are flagged as fine to delete.

It's definitely edge case stuff. that's why it's in the troubleshooting section.

Reply:
ah. reading this paragraph against it's not clear that's what I'm talking about. let me try an update.

Reply:
Take a look at the section with the rewrite and let me know if it's clearer, please.

Reply:
We first commit MOB followed by store file. If commit time of a MOB is less than choreStartTime - minAgeToArchive it can be safely archived if the following conditions are met:

1. Time skew across the cluster is less than  minAgeToArchive (default is 1h). 
2. Under no circumstances time difference between committing MOB and store file can exceed minAgeToArchive. Again - 1h. 

That is it, now we do not depend on Chore procedure duration as in v14 patch, b/c the algo was changed.